### PR TITLE
fix: make single guardian devimint cli test backwards-compatible

### LINKED ISSUE #4017: `devimint_cli_test_single` fails backwards-compatibility test
https://github.com/fedimint/fedimint/actions/runs/7434410505/job/20228534969?pr=4010#step:11:2879

```
00:00:17 thread 'main' panicked at devimint/src/main.rs:459:9:
00:00:17 assertion failed: descriptor.contains(\"wpkh(\")
```

### PR DESCRIPTION
**Pull request by bradleystachurski**
Fixes #4017 

I verified locally against the full matrix of backwards-compatibility tests that `devimint_cli_test_single` succeeds (see: https://github.com/fedimint/fedimint/pull/4010).

```
Backwards-compatibility tests summary:
fed_version  client_version  gateway_version  exit_code
v0.2.1       v0.2.1          current          0
v0.2.1       current         v0.2.1           0
v0.2.1       current         current          0
current      v0.2.1          v0.2.1           0
current      v0.2.1          current          0
current      current         v0.2.1           0
```

### CODE REVIEW

**Review by elsirion — DISMISSED** [review]
:tada:

File: devimint/src/main.rs [hunk 1]
```diff
@@ -416,19 +416,27 @@ async fn cli_tests(dev_fed: DevFed) -> Result<()> {
...
-        .as_str()
-        .unwrap()
-        .to_owned();
+    // fedimintd introduced wpkh for single guardian federations in v0.3.0 (9e35bdb)
+    // The code path is backwards-compatible, however this test will fail if we
+    // check against earlier fedimintd versions.
+    let fedimintd_version_res = cmd!(devimint::util::FedimintdCmd, "--version")
+        .out_string()
+        .await;
+    if devimint::util::is_min_version(0, 3, 0, fedimintd_version_res) {
```
Comment (dpc) [inline devimint/src/main.rs:425]: `is_min_version` confuses me. Would expect `is_version_at_least` or `is_version_gt` kind of things.

Also `0, 3, 0` ...  looks weird.

File: devimint/src/util.rs [hunk 2]
```diff
@@ -419,3 +420,153 @@ impl GatewayLndCli {
...
+        minor: captured["minor"].parse()?,
+        patch: captured["patch"].parse()?,
+    })
+}
+
+/// Returns true if the provided version string is greater than or equal to the
+/// min version provided. If there's an error parsing the version, returns
+/// false.
+pub fn is_min_version(
+    min_major: u32,
```
Comment (dpc) [inline devimint/src/util.rs:480]: If we have `CrateVersion` why not use it here?

File: devimint/src/util.rs [hunk 3]
```diff
@@ -419,3 +420,153 @@ impl GatewayLndCli {
...
+    min_minor: u32,
+    min_patch: u32,
+    version_res: Result<String, anyhow::Error>,
```
Comment (dpc) [inline devimint/src/util.rs:483]: Taking `Result<String, anyhow::Error>` here is odd and confusing.

If for some reason older versions didn't have `--version` or something like that, then the version parsing function should default to some magic number (0.2.0? 0.1.0?) to unify all the cases.

Comment (dpc) [reply devimint/src/util.rs:483]: Without these issues, we can use some more natural version comparison (I'm not sure if `cmp::Ordering` and operators like `<=` could be used for semver, need more coffee).

BTW. https://docs.rs/semver/latest/semver/ exists and [is not a heavy crate](https://crates.io/crates/semver/1.0.21/dependencies), maybe we should just use it?

Comment (bradleystachurski) [reply devimint/src/util.rs]: Good call 👍 

If we're comfortable introducing the semver crate, it simplifies this PR.

https://github.com/fedimint/fedimint/pull/4104/commits/2fa38846037d28a13348254f95eb4d0a12139257

**Review by dpc — APPROVED** [review]
:shipit:

**Review by elsirion — APPROVED** [review]

**Pull request merged**